db/document.go

```diff
@@ -811,3 +811,113 @@ func (doc *document) MarshalWithXattr() (data []byte, xdata []byte, err error) {
 
 	return data, xdata, nil
 }
+
+// Equal returns true if both of the given attachment metadata maps are considered the same.
+func (a AttachmentsMeta) Equal(b AttachmentsMeta) bool {
+	if len(a) != len(b) {
+		return false
+	} else if a == nil && b != nil {
+		return false
+	} else if b == nil && a != nil {
+		return false
+	}
+
+	// Find attachments in B that aren't in A
+	for bAttachmentName := range b {
+		if _, seen := a[bAttachmentName]; !seen {
+			return false
+		}
+	}
+
+	// Make sure attachments in A match those in B
+	for aAttachmentName, aAttachment := range a {
+		aAttachmentMap, ok := aAttachment.(map[string]interface{})
+		if !ok {
+			return false
+		}
+		bAttachmentMap, ok := b[aAttachmentName].(map[string]interface{})
+		if !ok {
+			return false
+		}
+
+		// Have to do really ugly type assertions/map checks because AttachmentsMeta is just a map[string]interface{}
+		// This would be a LOT cleaner if we used an AttachmentMap for DocAttachment structs in the revcache instead.
+		if aVal, ok := aAttachmentMap["content_type"]; !ok {
+			return false
```

Comment:
For the purposes of 'Equal' I don't think we should be validating whether fields are present .  If a and b both omitted content_type, they should still be equal.